nfcd/src/main/jni/src/helper/MapInfo.cpp

```diff
@@ -13,8 +14,16 @@ bool MapInfo::create() {
     int rv = dl_iterate_phdr([] (struct dl_phdr_info *info, size_t, void *user_data) {
         auto *instance = (MapInfo *)user_data;
 
+        // Skip entries with null names to prevent crashes, but keep empty names (main executable)
+        if (!info->dlpi_name) {
+            return 0;
+        }
+        
+        // Use a safe name for entries with empty names (typically main executable)
+        const char* libName = (strlen(info->dlpi_name) == 0) ? "<main>" : info->dlpi_name;
```

Comment:
Why is it a problem if the name is empty? The only issue that could occur is if multiple libraries map to the same (potentially empty) name and overwrite each other in the map. Maybe `dlpi_addr` could be used as an alternate name if the `dlpi_name` is empty?

Reply:
Good point. Impelmented it to use `dlpi_addr` as alternate name to avoid overwriting `<main>` for multiple unnamed mappings.